File: CPP/binarysearch.cpp
#include<iostream>
using namespace std;
int main(){
    int arr[5]={ 4 ,6 ,8 ,9 ,10};
    int  key ;
    cout<<"enter the key";
    cin>>key;
     int start=0;
     int end=4;
     int mid= start + (end-start)/2;
     while( start<=end){

            if( arr[mid] == key){
                return mid;

            }
            if( key< arr[mid]){
                end=mid-1;
            }
            else{
                start=mid+1;
            }
            mid= start + (end-start)/2;
     }
     return -1;
     
}

File: JAVA/binarysearch.java
public class binarysearch{
  public static void main(String[] args) {
    int[] array={1,2,3,4,5,6,7};
    int target= 1;
    // using binary search ->
    int value = BinarySearch(array,target);
    System.out.println(" The index at which the target is    "+value);
    
  }
   static int  BinarySearch(int[] array, int target){
    int start= 0;
    int end= array.length -1;
  
    
    while(start<=end){
      int mid =  start+(end-start)/2;
      if (target<array[mid]){
         end=mid-1;
      }
     else if(target>array[mid]) {
      start=mid+1;

     }
     else{
      // ans found->
      return mid;
     }
    }
    return -1;
  
  }
}